### PR TITLE
API Authenticaton

### PR DESCRIPTION
**Pull request by Nathan-Chon**
animation85

### CODE REVIEW

File: api-authentication/server/server.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      from "users"
+      where "username" = $1
+    `;
+    const params = [username];
+    const result = await db.query(sql, params);
+    const user = result.rows[0];
+    if (!user) {
+      throw new ClientError(401, 'invalid login');
+    }
+    const isMatching = argon2.verify(user.hashedPassword, password);
```
Comment (Zhiwutian) [inline api-authentication/server/server.js:60]: verify should be run with `await`

**Review by Zhiwutian — CHANGES_REQUESTED** [review]
one small change!

**Comment by Nathan-Chon** [comment]
animation85

**Review by Zhiwutian — APPROVED** [review]
👍🏼

**Pull request merged**